### DISCUSSION TITLE
Pass Path as an argument to hooks, validate, defaultValue, filterOptions(for relations) etc.

### DISCUSSION THREAD

**Discussion by AndoGhevian** [payloadcms/payload#2416] [Feature Requests & Ideas]
### Description
Sometimes for deep structured documents, even with depth 2, like **group->group**, **array->array**, it is required to get **context** of outer **array Item**, or group fields, but there is no way for that. We have though access to data of a document, but we also need a **path** to:
- Get to the outer **array item**, because its name ( index ) is dynamic.
- In case of Group Fields to write reusable **Field Schemas**

This request concerns _fieldHooks_, _validations_, _defaultValue_, _filterOptions_ etc.

**Comment by jmikrut** [comment]
Golden idea.

Gonna add this to our roadmap.  I think it's a great addition.

**Comment by JarrodMFlesch** [comment]
The validate function now has path https://github.com/payloadcms/payload/pull/11591.
